### PR TITLE
Template Pipeline: structural `ng-template`s, temporary var decls, `!important`, and misc tests

### PR DESCRIPTION
**Pull request by dylhunn**
We fix some notable issues:

- bindings were incorrectly being placed on structural `ng-template` elements
- incorrect temporary variable declaration sites
- removal of `!important` in host bindings

See individual commits for details.

Also, we enable a number of passing and nearly-passing tests.

### CODE REVIEW

File: packages/compiler-cli/test/compliance/test_cases/r3_view_compiler_template/nested_template_context_template.js [hunk 1]
```diff
@@ -53,8 +53,8 @@ function MyComponent_ul_0_Template(rf, ctx) {
 …
 consts: [
   [__AttributeMarker.Template__, "ngFor", "ngForOf"],
-  [__AttributeMarker.Bindings__, "title", "click", __AttributeMarker.Template__, "ngFor", "ngForOf"],
-  [__AttributeMarker.Bindings__, "title", "click"]
+  [__AttributeMarker.Bindings__, "click", "title", __AttributeMarker.Template__, "ngFor", "ngForOf"],
```
Comment (mmalerba) [inline packages/compiler-cli/test/compliance/test_cases/r3_view_compiler_template/nested_template_context_template.js:56]: I think you can revert this file

File: packages/compiler/src/template/pipeline/src/ingest.ts [hunk 2]
```diff
@@ -503,14 +520,43 @@ const BINDING_KINDS = new Map<e.BindingType, ir.BindingKind>([
...
+   * The binding is to a static text literal and not to an expression.
+   */
+  TextValue = 0b0001,
+
+  /**
+   * The binding belongs to the `<ng-template>` side of a `t.Template`.
+   */
+  BindingTargetsTemplate = 0b0010,
+
+  IsStructuralTemplateAttribute = 0b0100,
```
Comment (mmalerba) [inline packages/compiler/src/template/pipeline/src/ingest.ts:536]: nit: missing doc string

File: packages/compiler/src/template/pipeline/src/phases/temporary_variables.ts [hunk 3]
```diff
@@ -21,49 +21,69 @@ import {CompilationJob, ComponentCompilationJob} from '../compilation';
...
-            count--;
-          }
-          assignName(defs, expr);
+  // For each op, search for any variables that are assigned or read. For each variable, generate a
+  // name and produce a `DeclareVarStmt` to the beginning of the block.
+  for (const op of ops) {
+    // Identify the final time each temp var is read.
+    const finalReads = new Map<ir.XrefId, ir.ReadTemporaryExpr>();
+    ir.visitExpressionsInOp(op, (expr, flag) => {
+      if (flag === ir.VisitorContextFlag.InChildOperation) {
```
Comment (mmalerba) [inline packages/compiler/src/template/pipeline/src/phases/temporary_variables.ts:40]: Probably best to use `&`

File: packages/compiler/src/template/pipeline/src/phases/temporary_variables.ts [hunk 4]
```diff
@@ -21,49 +21,69 @@ import {CompilationJob, ComponentCompilationJob} from '../compilation';
...
+    });
+
+    // Name the temp vars, accounting for the fact that a name can be reused after it has been
+    // read for the final time.
+    let count = 0;
+    const assigned = new Set<ir.XrefId>();
+    const released = new Set<ir.XrefId>();
+    const defs = new Map<ir.XrefId, string>();
+    ir.visitExpressionsInOp(op, (expr, flag) => {
+      if (flag === ir.VisitorContextFlag.InChildOperation) {
```
Comment (mmalerba) [inline packages/compiler/src/template/pipeline/src/phases/temporary_variables.ts:55]: Same about `&`

File: packages/compiler/src/template/pipeline/src/phases/temporary_variables.ts (hunk 3 above)
Comment (dylhunn) [reply packages/compiler/src/template/pipeline/src/phases/temporary_variables.ts:40]: Whoops, nice catch!

**Comment by dylhunn** [comment]
Thanks @mmalerba, all comments resolved

**Review by mmalerba — APPROVED** [review]

**Comment by dylhunn** [comment]
caretaker: framework-win and aio-local are flakes

**Pull request closed without merging**